### PR TITLE
[cisco] enable and fortify ECN config GCU testing

### PR DESCRIPTION
**Pull request by dcaugher**
### Description of PR

Summary:
Fixes the case where a WRED_PROFILE cannot be retrieved from the CONFIG_DB. Also enables some Cisco HW SKUs for ECN testing, and fixes some corner cases for the test itself. (What if, for example, the original value was 99 or 100, in the original version?)

### Type of change

- [X] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311

#### What is the motivation for this PR?

To allow the ECN/GCU test to complete on Cisco platforms in a T2 topology to complete successfully

#### How did you do it?

Verified that the HW SKU was in fact supported. Skipped tests whenever a WRED_PROFILE could not be retrieved for a particular system.

#### How did you verify/test it?

Tested by changing the search string to FRED_PROFILE for a moment just to force empty values to be returned in some cases.
 
#### Any platform specific information?

See PR for specific HW SKU information

#### Supported testbed topology if it's a new test case?

T2

### CODE REVIEW

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 1]
```diff
@@ -2172,7 +2172,7 @@ generic_config_updater/test_ecn_config_update.py::test_ecn_config_updates:
     reason: "This test is not run on this asic type, topology, or version currently"
     conditions_logical_operator: "OR"
     conditions:
-      - "asic_type in ['cisco-8000']"
+      - "hwsku in ['Cisco-8800-LC-48H-C48']"
```
Comment (cyw233) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:2467]: oh, can this test only run on Q100 LC?

Comment (dcaugher) [reply tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:2467]: This test should pass on all other LCs except as stated. (We're tracking the exception internally, more later.)

**Comment by cyw233** [comment]
Hi @dcaugher, I can see this PR has many irrelevant changes after the rebase, can you fix that?

You can run the following to rebase again
```
# pull the latest sonic-mgmt:master
# checkout to dcaugher:dcaugher-cisco-ecn-config-check-master branch
git reset --hard <your_latest_commit>  # I believe it should be 699f4274c4263fa8c1edd937e13d95fc9430fce3
git rebase master 
# check git logs to make sure everything looks ok
git push -f
```

I tried from my side but looks like I don't have the permission to push to your branch

**Comment by dcaugher** [comment]
Ugh, my bad. Please try again.

**Review by auspham — APPROVED** [review]
